test/tools/cli/OutputFormat.java

```diff
@@ -0,0 +1,76 @@
+package tools.cli;
+
+import com.amazon.ion.IonWriter;
+import com.amazon.ion.system.IonBinaryWriterBuilder;
+import com.amazon.ion.system.IonTextWriterBuilder;
+
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Represents the different Ion output formats supported by the command line tools in this package.
+ */
+public enum OutputFormat implements IonWriterCreator {
+    /**
+     * Nicely spaced, 'prettified' text Ion.
+     */
+    PRETTY("pretty") {
+        @Override
+        public IonWriter createIonWriter(OutputStream outputStream) {
+            return IonTextWriterBuilder.pretty().build(outputStream);
+        }
+    },
+    /**
+     * Minimally spaced text Ion.
+     */
+    TEXT("text") {
+        @Override
+        public IonWriter createIonWriter(OutputStream outputStream) {
+            return IonTextWriterBuilder.standard().build(outputStream);
+        }
+    },
+    /**
+     * Compact, read-optimized binary Ion.
+     */
+    BINARY("binary") {
+        @Override
+        public IonWriter createIonWriter(OutputStream outputStream) {
+            return IonBinaryWriterBuilder.standard().build(outputStream);
+        }
+    },
+    EVENT("events"){
+        @Override
+        public IonWriter createIonWriter(OutputStream outputStream) {
+            return IonTextWriterBuilder.pretty().build(outputStream);
+        }
+    };
+
+    private static final Map<String, OutputFormat> FORMATS_BY_NAME;
+    static {
+        FORMATS_BY_NAME = new HashMap<>();
+        for (OutputFormat outputFormat : OutputFormat.values()) {
+            FORMATS_BY_NAME.put(outputFormat.getName(), outputFormat);
+        }
+    }
+
+    /**
+     * Finds the OutputFormat associated with the provided name.
+     * @param name  The name of the OutputFormat being requested.
+     * @return  OutputFormat or empty if the provided name is not associated with an OutputFormat.
+     */
+    public static Optional<OutputFormat> forName(String name) {
+        return Optional.ofNullable(FORMATS_BY_NAME.get(name));
+    }
+
+    private String name;
+
+    OutputFormat(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
```

Comment:
two follow-ups after deleting this:

1. Since we simply convert all letters of the string to upper case, "event" performances the same as "EVENTS" or "eVeNtS".

2. ValueOf(name.toUpperCase()) method will throw an IllegalArgumentException if the name is not exist in enum. 

To solve issue 1, I kept the name field and compare it with the parsed argument to make sure they are matched.(OutputFormat.java 53-59)

There are two ways to solve issue 2, catch IllegalArgumentException or create a help function to check if the name exists in enum. I created a function called nameToOutputFormat (OutputFormat.java 63-70) which checks if the name is valid and throw a CmdLineException if the name is invalid.

Reply:
I don't really see issue 1 as a problem. You can accept all capitalizations of the same word the same way. I think having a separate "name" field is redundant because enums already have this.

For issue 2, I think just allowing IllegalArgumentException to bubble up is fine. Your solution to handle the exception is also fine.

Reply:
Removed name field, and catch IllegalArgumentException in main now. (IonProcess.java 57)